### DISCUSSION TITLE
Sameday filter on Recent sidebar tab cost too much time on each type

### DISCUSSION THREAD

**Discussion by linonetwo** [TiddlyWiki/TiddlyWiki5#8488] [General]
![图片](https://github.com/user-attachments/assets/f0b8a8b9-abcc-4eb7-9e18-87dcaa0056d5)

exports.sameday @ sameday.js:18:27

So typing in tiddler isn't fun with 0.5s lagging.

There must be some way to fix this, async widget calculation in worker is what I mostly want, but requres heavy modification to the core.

So I think it might be easier to let sidebar Recent tab ignore draft tiddler's change.

**Comment by Jermolene** [comment]
Hi @linonetwo I think the first thing to do might be to attempt some optimisations to the imlementation of "sameday".

Currently, it converts the target operand to a JS date, and then converts each input title to a date for the comparison. Creating objects like that is expensive. In this case it should be possible to avoid the conversion completely, and perform the calculation based on the raw `YYYYMMDDHHMMSSXXX` notation.
